### PR TITLE
[Electron enhancement] Window Handle for electron application for multiple execution

### PR DESCRIPTION
**Pull request by MedhaGupta-Ril**
-Added new two-parameter for electron application.
-One Page Load timeout and one loading page is available for the application to handle windows.

### CODE REVIEW

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java [hunk 1]
```diff
@@ -587,11 +581,32 @@ static Driver createElectronDriverForUser(String userPersona, String browserName
                         : ((ChromeDriver) driver).getCapabilities();
         Drivers.addUserPersonaDriverCapabilities(userPersona, capabilities);
         LOGGER.info("Electron driver capabilities extracted for further use");
-        loadBaseUrl(baseUrl, driver);
+
+        if (browserConfigForBrowserType.getBoolean("electronAppLoadingPage"))
+            handleWindowForElectronApplication(driver, browserConfigForBrowserType);
```
Comment (anandbagmar) [inline src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:586]: add {}

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java [hunk 2]
```diff
@@ -587,11 +581,32 @@ static Driver createElectronDriverForUser(String userPersona, String browserName
...
         Driver currentDriver = new Driver(updatedTestName, forPlatform, userPersona, appName, driver, isRunInHeadlessMode);
         numberOfWebDriversUsed++;
         LOGGER.info(String.format("createElectronDriverForUser: done: userPersona: '%s', Platform: '%s', appName: '%s', Number of ElectronDrivers: '%d'",
                 userPersona, forPlatform.name(), appName, numberOfWebDriversUsed));
         return currentDriver;
     }
+
+    private static void handleWindowForElectronApplication(WebDriver driver, JSONObject browserConfigForBrowserType) {
+        Set<String> windowHandles = new HashSet<>();
+        String parentWindowHandle = driver.getWindowHandle();
```
Comment (anandbagmar) [inline src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:597]: Add relevant log messages

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java [hunk 3]
```diff
@@ -587,11 +581,32 @@ static Driver createElectronDriverForUser(String userPersona, String browserName
...
+        long startTime = System.currentTimeMillis();
+
+        while (windowHandles.size() <= 1) {
+            windowHandles = driver.getWindowHandles();
+            if(System.currentTimeMillis() - startTime > browserConfigForBrowserType.getInt("electronAppLoadTime") * 1000L)
+                throw new TimeoutException("Only one window handle is available");
+        }
+        for (String handle : windowHandles) {
+            if (!handle.equals(parentWindowHandle)) {
+                driver.switchTo().window(handle);
```
Comment (anandbagmar) [inline src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:607]: add log message when switching windows

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java [hunk 4]
```diff
@@ -14,11 +14,7 @@
 import org.jetbrains.annotations.NotNull;
 import org.json.JSONArray;
 import org.json.JSONObject;
-import org.openqa.selenium.Capabilities;
-import org.openqa.selenium.Dimension;
-import org.openqa.selenium.MutableCapabilities;
-import org.openqa.selenium.Proxy;
-import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.*;
```
[the remark this answers is not in the document]
Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:17]: Updated

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java (hunk 1 above)
Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:586]: Updated

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java (hunk 2 above)
Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:605]: Updated

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java (hunk 3 above)
Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:618]: Updated

File: src/main/java/com/znsio/teswiz/runner/Visual.java [hunk 5]
```diff
@@ -297,15 +298,14 @@ private RectangleSize getBrowserViewPortSize(String driverType, WebDriver innerD
...
             JavascriptExecutor js = (JavascriptExecutor) innerDriver;
-            if (Runner.getPlatform().equals(Platform.electron)) {
-                Set<String> windowHandles = innerDriver.getWindowHandles();
-                if (windowHandles.size() > 0) {
-                    innerDriver.switchTo().window((String) windowHandles.toArray()[0]);
-                }
-            } else {
-                Dimension actualBrowserSize = innerDriver.manage().window().getSize();
-                LOGGER.info(String.format("Actual browser dimensions: %s", actualBrowserSize));
+            Dimension actualBrowserSize;
```
[the remark this answers is not in the document]
Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/runner/Visual.java:301]: Not required, adding a condition to set the dimension for electron application window size, logger is already added for window size. after if-else condition.

File: src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java [hunk 6]
```diff
@@ -35,9 +36,7 @@
 import java.io.File;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (anandbagmar) [inline src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java:39]: single class imports

**Pull request merged**